contracts/peripheral/GaugeVoteRewards.sol

```diff
@@ -0,0 +1,292 @@
+        stashRewardDistro = IStashRewardDistro(_stashRewardDistro);
+        lzChainId = _lzChainId;
+
+        _initializeLzApp(_lzEndpoint);
+
+        // Approve AuraOFT with AURA
+        IERC20(_aura).safeApprove(_auraOFT, type(uint256).max);
+    }
+
+    /* -------------------------------------------------------------------
+       Modifiers 
+    ------------------------------------------------------------------- */
+
+    modifier onlyDistributor() {
+        require(msg.sender == distributor, "!distributor");
+        _;
+    }
+
+    /* -------------------------------------------------------------------
+       Setters 
+    ------------------------------------------------------------------- */
+
+    function setDistributor(address _distributor) external onlyOwner {
+        distributor = _distributor;
+        emit SetDistributor(_distributor);
+    }
+
+    function setRewardPerEpoch(uint256 _rewardPerEpoch) external onlyOwner {
+        rewardPerEpoch = _rewardPerEpoch;
+        emit SetRewardPerEpoch(_rewardPerEpoch);
+    }
+
+    function setPoolIds(uint256[] memory _poolIds) external {
+        uint256 poolIdsLen = _poolIds.length;
+        for (uint256 i = 0; i < poolIdsLen; i++) {
+            uint256 pid = _poolIds[i];
+            IBooster.PoolInfo memory poolInfo = booster.poolInfo(pid);
+            getPoolId[poolInfo.gauge] = pid;
+        }
+    }
+
+    function setDstChainId(address[] memory _gauges, uint16[] memory _dstChainIds) external onlyOwner {
+        uint256 dstChainIdsLen = _dstChainIds.length;
+        for (uint256 i = 0; i < dstChainIdsLen; i++) {
+            getDstChainId[_gauges[i]] = _dstChainIds[i];
+        }
+    }
+
+    function setChildGaugeVoteRewards(uint16[] memory _dstChainIds, address[] memory _voteRewards) external onlyOwner {
+        uint256 dstChainIdsLen = _dstChainIds.length;
+        require(dstChainIdsLen == _voteRewards.length, "!length");
+        for (uint256 i = 0; i < dstChainIdsLen; i++) {
+            getChildGaugeVoteRewards[_dstChainIds[i]] = _voteRewards[i];
+        }
+    }
+
+    /* -------------------------------------------------------------------
+       View 
+    ------------------------------------------------------------------- */
+
+    function getCurrentEpoch() external view returns (uint256) {
+        return _getCurrentEpoch();
+    }
+
+    /* -------------------------------------------------------------------
+       Core 
+    ------------------------------------------------------------------- */
+
+    /**
+     * @notice  Wraps the booster.voteGaugeWeight call to track weights for each epoch
+     *          So AURA rewards can be distributed pro rata to those pool stashes
+     * @param _gauge    Array of the gauges
+     * @param _weight   Array of the weights
+     * @return bool for success
+     */
+    function voteGaugeWeight(address[] calldata _gauge, uint256[] calldata _weight) external onlyOwner returns (bool) {
```

Comment:
It can be called more than once, the restrictions will be set in the GaugeController that wont let you vote for more than 10k weight. So there is no need to add any additional checks here.